File: app/src/main/java/ru/MeatGames/roguelike/tomb/Global.java
package ru.MeatGames.roguelike.tomb;

import ru.MeatGames.roguelike.tomb.model.HeroClass;
import ru.MeatGames.roguelike.tomb.model.MapClass;
import ru.MeatGames.roguelike.tomb.screen.BrezenhamView;
import ru.MeatGames.roguelike.tomb.screen.InventoryView;
import ru.MeatGames.roguelike.tomb.screen.MainMenu;
import ru.MeatGames.roguelike.tomb.screen.MapView;
import ru.MeatGames.roguelike.tomb.screen.StatsView;
import ru.MeatGames.roguelike.tomb.util.AssetHelper;

public class Global {

    public static Game game;
    public static HeroClass hero;
    public static MapGenerationClass mapg;
    public static MapView mapview;
    public static InventoryView invview;
    public static StatsView stsview;
    public static BrezenhamView bview;
    public static MainMenu mmview;
    public static MapClass[][] map;
    public static TileDB[] tiles; // 0 element is opaque
    public static ItemDB[] itemDB;
    public static MobDB[] mobDB;
    public static StatsDB[] stats;
    public static AssetHelper mAssetHelper;

}

File: app/src/main/java/ru/MeatGames/roguelike/tomb/Global.kt
package ru.MeatGames.roguelike.tomb

import android.os.Vibrator
import ru.MeatGames.roguelike.tomb.db.ItemDB
import ru.MeatGames.roguelike.tomb.db.MobDB
import ru.MeatGames.roguelike.tomb.db.StatsDB
import ru.MeatGames.roguelike.tomb.db.TileDB
import ru.MeatGames.roguelike.tomb.model.HeroClass
import ru.MeatGames.roguelike.tomb.model.MapClass
import ru.MeatGames.roguelike.tomb.screen.GameScreen
import ru.MeatGames.roguelike.tomb.screen.MainMenu
import ru.MeatGames.roguelike.tomb.util.AssetHelper

object Global {

    lateinit var game: Game

    var hero: HeroClass? = null
    var map: Array<Array<MapClass>>? = null

    lateinit var mapview: GameScreen
    lateinit var mmview: MainMenu

    // temporary most of the tiles is null, use carefully
    lateinit var tiles: Array<TileDB> // 0 element is opaque
    lateinit var itemDB: Array<ItemDB>
    lateinit var mobDB: Array<MobDB>
    lateinit var stats: Array<StatsDB>

    lateinit var mAssetHelper: AssetHelper

    lateinit var mVibrator: Vibrator

    @JvmStatic
    fun vibrate() {
        mVibrator.vibrate(30)
    }

}